aws/resource_aws_codepipeline_webhook.go

```diff
@@ -0,0 +1,336 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/codepipeline"
+)
+
+func resourceAwsCodePipelineWebhook() *schema.Resource {
+	return &schema.Resource{
+		Create:        resourceAwsCodePipelineWebhookCreate,
+		Read:          resourceAwsCodePipelineWebhookRead,
+		Update:        nil,
+		Delete:        resourceAwsCodePipelineWebhookDelete,
+		SchemaVersion: 1,
+
+		Schema: map[string]*schema.Schema{
+			"authentication": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					codepipeline.WebhookAuthenticationTypeGithubHmac,
+					codepipeline.WebhookAuthenticationTypeIp,
+					codepipeline.WebhookAuthenticationTypeUnauthenticated,
+				}, false),
+			},
+			"authentication_configuration": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				MinItems: 1,
+				Required: true,
```

Comment:
Is this required for `authentication = "UNAUTHENTICATED"`?

Reply:
It's not. I'll mark it as `Optional:  true` instead.